File: pset2/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <ctype.h>
#include <string.h>
#include <math.h>

int main(void)
{
//ask for the string
    string text = get_string(" Text: ");

//count number of letters
    float letterCount = 0;

//for each character add one if it is a letter
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (isalpha(text[i]))
        {
            letterCount ++;
        }
    }

//count number of words
    float wordCount = 1;

//for each character add one if it is a word
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (isblank(text[i]) && isalpha(text[i + 1]))

//rediculous number of comments
        {
            wordCount ++;
        }

//adds a word even after ""
        else if (isblank(text[i]) && text[i + 1] == '"')
        {
            wordCount ++;
        }
    }
//this is bonkers comment

//count number of sentences
    float sentenceCount = 0;
//I really cant add more commients
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (text[i] == ('.') || text[i] == ('?') || text[i] == ('!'))
        {
            sentenceCount++;
        }
    }

//calculate the grade
//letters per word *100
    float L = (letterCount / wordCount) * 100;

//number of sentences per 100 words
    float S = ((sentenceCount / wordCount) * 100);

//calculate index
    float index = ((0.0588 * L) - (0.296 * S) - 15.8);

//output grade
    int grade = round(index);
//output for less than 1
    if (grade < 1)
    {
        printf("Before Grade 1\n");
    }

//output for more than 16
    else if (grade >= 16)
    {
        printf("Grade 16+\n");
    }

//main output
    else
    {
        printf("Grade %i\n", grade);
    }
}

File: pset6/readability/readability.py
from cs50 import get_string
import string


# ask for the string
text = get_string("Text: ")

# count number of letters

lettersOnly = [c.lower() for c in text if c.isalpha()]
letterCount = len(lettersOnly)

# count number of words
wordCount = len(text.split())

# count sentences
fullstopCount = text.count(".")
exclaimCount = text.count("!")
questionCount = text.count("?")
sentenceCount = (fullstopCount + exclaimCount + questionCount)

# letters per word *100
L = ((letterCount / wordCount) * 100)

#  of sentences per 100 words
S = ((sentenceCount / wordCount) * 100)

# calculate index
index = ((0.0588 * L) - (0.296 * S) - 15.8)

# output grade
grade = round(index)

# output for less than 1
if (grade < 1):
    print("Before Grade 1\n")

# output for more than 16
elif (grade >= 16):
    print("Grade 16+\n")

# main output
else:
    print(f"Grade {grade}")